Adding lfric_reader

tinyendian opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team][1] using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel][2] if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there. (I am currently the only maintainer)
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/lfric_reader```) and found some lint.

Here's what I've got...


For **recipes/lfric_reader**:

 * If python is a host requirement, it should be a run requirement.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/lfric_reader```) and found it was in an excellent condition.

tinyendian commented:
@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

tinyendian commented:
Reade for review, thanks

tinyendian commented:
@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

bjlittle commented:
@ocefpaf Could I call in a favour and ask for a merge to unblock this PR? 😉 

I'm collaborating with colleagues at NIWA, and they'd greatly appreciate having their package on `conda-forge` 👍

xhochy commented on recipes/lfric_reader/build.sh:
```diff
@@ -0,0 +1,16 @@
+cd src/cxx
+cmake CMakeLists.txt \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DBUILD_SHARED_LIBS=ON \
+      -DBUILD_TESTING=OFF
```
We have some common CMake arguments, please use them.
```suggestion
cmake ${CMAKE_ARGS} CMakeLists.txt \
      -DCMAKE_BUILD_TYPE=Release \
      -DBUILD_SHARED_LIBS=ON \
      -DBUILD_TESTING=OFF
```

tinyendian replied:
Implemented in commit below, thanks

xhochy commented on recipes/lfric_reader/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libxext') }}     # [linux]
+    - cmake
+    - boost
+    - eigen
```
These probably belong into `host`

tinyendian replied:
Implemented in commit below, thanks

tinyendian commented:
@xhochy thanks a lot for the review - I implemented all suggested changes. I also added an additional CMake flag that specifies Python version explicitly, after the second package build during the automated GitHub Actions check on Linux suddenly broke, with CMake being unable to find Python.

ocefpaf commented:
LGTM. @xhochy got here for a review first so I'll let him merge it if he is OK with the changes.
